nodes/basic-pow/src/service.rs

```diff
@@ -193,6 +196,33 @@ pub fn new_full(config: Configuration) -> Result<TaskManager, ServiceError> {
 		task_manager
 			.spawn_essential_handle()
 			.spawn_blocking("pow", worker_task);
+		
+		// Start Mining
+		let mut numb: u8 = 0;
+		thread::spawn(move || {
+			loop {
+				let worker = _worker.clone();
+				let metadata = worker.lock().metadata();
+				if let Some(metadata) = metadata {
+					let nonce = H256::from_slice(Sha3_256::digest(&[numb]).as_slice());
+					let compute = Compute {
+						difficulty: metadata.difficulty,
+						pre_hash: metadata.pre_hash,
+						nonce,
+					};
+					let seal = compute.compute();
+					if hash_meets_difficulty(&seal.work, seal.difficulty) {
+						let mut worker = worker.lock();
+						worker.submit(seal.encode());
+					}
+					numb = numb.saturating_add(1u8);
+					if numb == 255u8 {
+						numb = 0;
+					}
+					thread::sleep(Duration::new(0, 500_000_000));
```

Comment:
Isn't the logic here should be something like this?

```suggestion
					if hash_meets_difficulty(&seal.work, seal.difficulty) {
						let mut worker = worker.lock();
						worker.submit(seal.encode());
					} else {
					  numb = numb.saturating_add(1u8);
					  if numb == 255u8 {
						  numb = 0;
					  }
					}
```

The last line of `thread::sleep` is in the `else` branch of `if let Some(metadata) = metadata {..} else { thread::sleep }`, based on code structure in https://github.com/kulupu/kulupu/blob/master/src/service.rs#L330-L377

Reply:
putting the `thread::sleep` in the `else` branch is for producing a delay in case no Block is yet ready to start mining, but i added it to create a delay in the mining process itself, since the `difficulty` was reduced due to the usage of the `u8` numb variable to generate a nonce.
i updated this, the nonce uses the type `U256` now.